src/Mod/Sketcher/Gui/CommandSketcherTools.cpp

```diff
@@ -2522,3 +2529,27 @@ void CreateSketcherCommandsConstraintAccel()
     rcCmdMgr.addCommand(new CmdSketcherPaste());
 }
 // clang-format on
+
+void SketcherGui::centerScale(Sketcher::SketchObject* Obj, double scale_factor)
+{
+    std::vector<int> allGeoIds(Obj->Geometry.getValues().size());
+    std::iota(allGeoIds.begin(), allGeoIds.end(), 0);
+
+
+    auto doc = Gui::Application::Instance->activeDocument();
+    SketcherGui::ViewProviderSketch* vp =
+        static_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+    auto scaler = DrawSketchHandlerScale::make_centerScale(allGeoIds, scale_factor, false);
+    scaler->setSketchGui(vp);
+    scaler->executeCommands();
+
+    auto view3d = dynamic_cast<Gui::View3DInventor*>(doc->getActiveView());
+    if (view3d) {
```

Comment:
```suggestion
    if (auto* view3d = dynamic_cast<Gui::View3DInventor*>(doc->getActiveView())) {
```

It is good to place such things directly in condition as it saves a line of code and makes it impossible to use the variable outside of checked scope.